File: code/loop.py
import timeit

code = """
total = 0
for i in range(1000):
    for j in range(1000):
        total += i
print total
"""
t = timeit.Timer(code)
dur = t.timeit(number=1)
print "ran in: {0:.3f} sec".format(dur)


# manual optimization
code = """
print sum( [1000 * i for i in range(1000)] )
"""
t = timeit.Timer(code)
dur = t.timeit(number=1)
print "ran in: {0:.3f} sec".format(dur)

File: code/loop.c
#include <stdio.h>
#include <time.h>

/* compile with: gcc -O2 -o loop loop.c */

int main() {
    clock_t f = 0;
    clock_t s = clock();

    long total = 0;
    for (int i = 0; i < 1000; i++)
        for (int j = 0; j < 1000; j++) {
            total += i;
        }

    printf("%ld\n", total);
    f = clock();
    printf("ran in: %.3f sec\n", ((double)f - (double)s) * 1.0e-6);
    return 0;
}

